packages/venia-concept/src/components/App/app.js

```diff
@@ -6,17 +6,51 @@ import Mask from 'src/components/Mask';
 import MiniCart from 'src/components/MiniCart';
 import Navigation from 'src/components/Navigation';
 import OnlineIndicator from 'src/components/OnlineIndicator';
+import ErrorNotifications from './errorNotifications';
 import renderRoutes from './renderRoutes';
+import errorRecord from 'src/util/createErrorRecord';
 
 class App extends Component {
     static propTypes = {
         app: shape({
             drawer: string,
             overlay: bool.isRequired
         }).isRequired,
-        closeDrawer: func.isRequired
+        closeDrawer: func.isRequired,
+        markErrorHandled: func.isRequired,
+        hasBeenOffline: bool,
+        isOnline: bool
     };
 
+    static get initialState() {
+        return {
+            renderError: null
+        };
+    }
+
+    get errorFallback() {
+        const { renderError } = this.state;
+        if (renderError) {
+            return (
+                <Fragment>
+                    <Main isMasked={true} />
+                    <Mask isActive={true} />
+                    <ErrorNotifications
+                        errors={[
+                            errorRecord(
+                                renderError,
+                                window,
+                                this,
+                                renderError.stack
+                            )
+                        ]}
```

Comment:
Consider calling `errorRecord()` outside of the JSX tree to preserve the tree structure.

```jsx
const record = errorRecord()

<ErrorNotifications errors={[record]} />
```

Also, won't this create a new array on every render (as long as `renderError` is still truthy across renders)?

Reply:
Yes, because in the event of a render error, there will always be only one, and it should replace any existing errors from Redux action flow. This is as it should be.